### PR TITLE
Pattern expression classes

### PR DESCRIPTION
**Pull request by rpiazza**

### CODE REVIEW

**Comment by gtback** [comment]
I think this depends on the new version of python-stix2, which we haven't pushed to PyPI yet.

File: stix2elevator/test/test_idioms.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def idiom_mappings(xml_file_path, stored_json):
     if not get_option_value("policy") == "no_policy":
         print("'no_policy' is not allowed for testing")
     set_option_value("policy", "no_policy")
-
+    sys.setrecursionlimit(5000)
```
Comment (gtback) [inline stix2elevator/test/test_idioms.py]: This is already defined in `setup_tests` (recently merged into master)

File: idioms-json/file-and-directory-pattern.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
                 "ip-watchlist"
             ],
             "modified": "2014-05-08T09:00:00.000Z",
-            "pattern": "[(file:hashes.'SHA-256' = 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855' AND (file:file_name = 'iprip32.dll' AND file:parent_directory_ref.path = '/usr/local'))]",
```
Comment (gtback) [inline idioms-json/file-and-directory-pattern.json]: Do all of the file_name->name changes need to be in this branch? If not I'd rather keep them separate.

File: stix2elevator/cli.py [hunk 3]
```diff
@@ -170,6 +170,8 @@ def main():
     elevator_args = elevator_parser.parse_args()
 
     initialize_options(elevator_args)
+    print(sys.getrecursionlimit())
```
Comment (gtback) [inline stix2elevator/cli.py]: I'm not sure if we want to set the recursion limit in our CLI, but I can see arguments both ways. If we do, maybe we can set it using a command-line option? I don't think we need to print it out though ;-)

File: stix2elevator/test/test_idioms.py (hunk 1 above)
Comment (rpiazza) [reply stix2elevator/test/test_idioms.py]: removed

File: stix2elevator/cli.py (hunk 3 above)
Comment (rpiazza) [reply stix2elevator/cli.py]: Maybe add a option on the command line?

File: stix2elevator/test/test_idioms.py (hunk 1 above)
Comment (gtback) [reply stix2elevator/test/test_idioms.py]: Did you push these changes to GitHub? I'm still seeing this line in the pattern-expression-classes branch

File: stix2elevator/cli.py (hunk 3 above)
Comment (gtback) [reply stix2elevator/cli.py]: I  think we can just remove this for now. If we need to, we can add an option later

**Pull request merged**